store.go

```diff
@@ -582,6 +593,7 @@ type store struct {
 	autoNsMaxSize   uint32
 	graphDriver     drivers.Driver
 	layerStore      LayerStore
+	roLayerStore    ROLayerStore
 	roLayerStores   []ROLayerStore
```

Comment:
Suppose the user calls:
- `s.Names(foo)` // Populates `roLayerStore`, reads from disk
- `s.PutLayer(…)` // Populates `layerStore`, reads + writes to disk
- `s.Names(foo)`

What happens now? AFAICS:
- `s.graphLock.TouchedSince` is false (no `.Shutdown*`), so the cached `roLayerStore` is used
- The RO and RW layer stores share lock objects ( https://github.com/containers/storage/blob/9f889830f10075aceb6b925a97927e77b837d0a7/pkg/lockfile/lockfile.go#L92-L100 ), so regardless of any `.Touch()` calls on locks, `.Modified()` will be `false`
- The RO layer store will continue reading from its cached data, regardless of modifications by the RW store.

(This is only by reading code I’m unfamiliar with, so I may well be missing something or making a stupid mistake.)